javav2/usecases/creating_redshift_application/Readme.md

```diff
@@ -0,0 +1,1199 @@
++ **WebSecurityConfig** - The role of this class is to set up an in-memory user store that contains a single user (the user name is **user** and the password is **password**).
+
+### BlogApp class
+
+The following Java code represents the **BlogApp** class.
+
+     package com.aws.blog;
+
+     import org.springframework.boot.SpringApplication;
+     import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+     @SpringBootApplication
+     public class BlogApp {
+
+     public static void main(String[] args) throws Throwable {
+        SpringApplication.run(BlogApp.class, args);
+      }
+     }
+
+### BlogController class
+
+The following Java code represents the **BlogController** class.
+
+     package com.aws.blog;
+
+     import org.springframework.security.core.context.SecurityContextHolder;
+     import org.springframework.stereotype.Controller;
+     import org.springframework.ui.Model;
+     import org.springframework.web.bind.annotation.GetMapping;
+     import org.springframework.web.bind.annotation.RequestMapping;
+     import org.springframework.web.bind.annotation.ResponseBody;
+     import org.springframework.web.bind.annotation.RequestMethod;
+     import javax.servlet.http.HttpServletRequest;
+     import javax.servlet.http.HttpServletResponse;
+     import org.springframework.beans.factory.annotation.Autowired;
+
+    @Controller
+    public class BlogController {
+
+    @Autowired
+    RedshiftService rs;
+
+    @GetMapping("/")
+    public String root() {
+        return "index";
+    }
+
+    @GetMapping("/add")
+    public String add() {
+        return "add";
+    }
+
+
+    @GetMapping("/posts")
+    public String post() {
+        return "post";
+    }
+
+    @GetMapping("/login")
+    public String login(Model model) {
+        return "login";
+    }
+
+    // Adds a new item to the database.
+    @RequestMapping(value = "/addPost", method = RequestMethod.POST)
+    @ResponseBody
+    String addItems(HttpServletRequest request, HttpServletResponse response) {
+
+        String name = getLoggedUser();
+        String title = request.getParameter("title");
+        String body = request.getParameter("body");
+        return rs.addRecord(name, title, body);
+    }
+
+
+    // Queries items from the Redshift database.
+    @RequestMapping(value = "/getPosts", method = RequestMethod.POST)
+    @ResponseBody
+    String getFivePosts(HttpServletRequest request, HttpServletResponse response) {
+
+        String num = request.getParameter("number");
+        String lang = request.getParameter("lang");
+        return rs.getPosts(lang,Integer.parseInt(num)) ;
+     }
+
+    private String getLoggedUser() {
+
+        org.springframework.security.core.userdetails.User user2 = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
Break long lines at approximately 80 characters so you don't have to scroll when reading.